tests/CMakeLists.txt

```diff
@@ -39,6 +39,13 @@ if(BSD_OVERLAY_FOUND)
                           ${BSD_OVERLAY_LDFLAGS})
 endif()
 
+if(CMAKE_SWIFT_RUNTIME_LIBDIR)
+  target_link_libraries(bsdtestharness
+                        PRIVATE
+                        -L${CMAKE_SWIFT_RUNTIME_LIBDIR} -lswiftCore -lswiftSwiftOnoneSupport
+                        -Wl,-rpath -Wl,${CMAKE_SWIFT_RUNTIME_LIBDIR})
```

Comment:
Adding rpaths is discouraged.  It can be used as a security issue to interposition other libraries.  I think think that we should be trying to do this differently.  Furthermore, the `-l` should not be added when doing the `target_link_libraries` as it will normally put the right prefix.  Finally, `-L` should be handled via `link_directories`.  Ill put up a PR to address some of this.